### DISCUSSION TITLE
There seems to be only one app:: Channel

### DISCUSSION THREAD

**Discussion by dongjian** [fltk-rs/fltk-rs#983] [General]
**Is your feature request related to a problem? Please describe.**
```
    let (s1, r1) = app::channel::<Message>();
    let (s2, r2) = app::channel::<Message>();
    let (s, r) = app::channel::<Message>();

    but_play.emit(s, Message::Test);
    but_stop.emit(s1, Message::Test);
    snap_shot.emit(s2, Message::Test);

    while app.wait() {
        if let Some(res) = r1.recv() {
            println!("r1: {:?}", res)
        }
        if let Some(res) = r2.recv() {
            println!("r2: {:?}", res)
        }
        if let Some(msg) = r.recv() {
            println!("r : {:?}", msg)
        }
}
```
and log 
r1: Test
r1: Test
r1: Test
If I modify the code:
 ```
    let (s1, r1) = app::channel::<i32>();
    let (s2, r2) = app::channel::<i32>();
    let (s, r) = app::channel::<Message>();
```
only i32 will be logged out, and all other messages will remain None, but the code will compile smoothly.

So I can only use one enum for the whole project

**Describe the solution you'd like**
Behavior similar to  std::sync::mpsc::channel

**Comment by MoAlyousef** [comment]
It's a known limitation of fltk channels, since they pipe to the same address space, and fixing to a single type ensures type safety. Providing wrappers against all channel types is out of the scope of this crate.

However, you can still use a different channel (standard channel or crossbeam-channel) if you need different behavior:
```rust
let (s, r) = std::sync::mpsc::channel::<Message>();
but_play.set_callback(move |_| {
    s.send(Message::Test).unwrap();
});
```

You can also define a method which applies to all widgets:
```rust
use std::sync::mpsc::Sender;

pub trait SenderWidget<W, T>
where
    W: WidgetExt,
    T: Send + Sync + Clone + 'static,
{
    fn send(&mut self, sender: Sender<T>, msg: T);
}

impl<W, T> SenderWidget<W, T> for W
where
    W: WidgetExt,
    T: Send + Sync + Clone + 'static,
{
    fn send(&mut self, sender: Sender<T>, msg: T) {
        self.set_callback(move |_| {
            sender.send(msg.clone()).unwrap();
        });
    }
}

fn main() {
    let (s, r) = std::sync::mpsc::channel::<Message>();
    but_play.send(s.clone(), Message::Test);
}
```

**Reply by dongjian** [reply]
Thanks for your explanation. I need to send a message in the VLC progress change callback. So use std channel app.wait() will not update. Now I use `sender.send(Message::DjVlcMessage(DjVlcMessage::ProgressUpdate(p)));`to solve this problem. Thank you for your help.
